lib/xboxrt/libc_extensions/wchar.h

```diff
@@ -55,6 +58,21 @@ int mbsinit( const mbstate_t * ps );
 size_t mbrtowc( wchar_t * XBOXRT_RESTRICT pwc, const char * XBOXRT_RESTRICT s, size_t n, mbstate_t * XBOXRT_RESTRICT ps );
 size_t wcrtomb( char * XBOXRT_RESTRICT s, wchar_t wc, mbstate_t * XBOXRT_RESTRICT ps );
 
+wchar_t *wmemset(wchar_t *wcs, wchar_t wc, size_t n);
+static double wcstod(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr) {assert(0); return 0.0;};
+static float wcstof(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr) {assert(0); return 0.0;};
+static long double wcstold(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr) {assert(0); return 0.0;};
+static long wcstol(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr, int base) {assert(0); return 0.0;};
+static long long wcstoll(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr, int base) {assert(0); return 0.0;};
+static unsigned long wcstoul(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr, int base) {assert(0); return 0.0;};
+static unsigned long long wcstoull(const wchar_t * XBOXRT_RESTRICT nptr, wchar_t ** XBOXRT_RESTRICT endptr, int base) {assert(0); return 0.0;};
+static size_t mbrlen(const char *s, size_t n, mbstate_t *ps) {assert(0); return 0;};
+static wint_t btowc(int c) {assert(0); return 0;};
+static size_t mbsrtowcs(wchar_t *dest, const char **src, size_t len, mbstate_t *ps) {assert(0); return 0;};
+static int wctob(wint_t c) {assert(0); return 0;};
+static int _snwprintf(wchar_t *buffer, size_t count, const wchar_t *format, ...) {assert(0); return 0;};
+static int wcrtomb_s(size_t * XBOXRT_RESTRICT retval, char * XBOXRT_RESTRICT s, size_t ssz, wchar_t wc, mbstate_t * XBOXRT_RESTRICT ps) {assert(0); return 0;};
```

Comment:
Why are some stubs in this header, while other stubs are in the .c file?
I'd probably prefer all stubs in the .c file, so we can avoid future back-and-forth changes in the header (and possible merge conflicts when touching nearby functions in different PRs).

Reply:
All declarations I added had their definition included in `wchar.h` as well. I moved them over to `wchar.c` now.

For `strtof` / `strtod` there was already an upstream header, which is not the case here yet. I'm not sure whether there's upstream interest to already have that header (but without implementations), but it's a bit outside the scope of what I'm wanting to work on rn.

Reply:
We should create an issue upstream and call out this discrepancy.

Reply:
This is actually somewhat consistent with https://rootdirectory.ddns.net/dokuwiki/doku.php?id=pdclib:start as stdlib is considered "DONE" but wchar.h hasn't even been started yet.